src/plugins/instrument_fetch.js

```diff
@@ -252,6 +256,27 @@ class InstrumentFetch {
         }
         return include;
     }
+
+    _shouldAddHeadersToRequest(tracer, url) {
+        // This shouldn't be possible, but let's be paranoid
+        if (!tracer || !url) {
+            return false;
+        }
+
+        let opts = tracer.options();
+        if (opts.disabled) {
+            return false;
+        }
+
+        let include = false;
+        if (opts.fetch_url_header_inclusion_patterns.some((inc) => inc.test(url))) {
+            include = true;
+        }
+        if (opts.fetch_url_header_exclusion_patterns.some((ex) => ex.test(url))) {
```

Comment:
this `if` statement for `exclusion` could be first condition and then `return false` without variable `include`
then next condition for `inclusion` and `return true` without variable
and as last simply `return false`.
no need for variable and I think there is no need to check for inclusion before exclusion as exclusion is more important

Reply:
Yea definitively agree. This was copy/pasted from the the way the other exclusion/inclusion logic works. I'll update both since I agree this logic is kinda wonky.